### DISCUSSION TITLE
11RC1.1 LayoutManager.ExecuteInitialLayoutPass

### DISCUSSION THREAD

**Discussion by msneijders** [AvaloniaUI/Avalonia#11625] [Q&A] [answered]
In a custom control (a ContentDialog like control), I am using `topLevel.LayoutManager.ExecuteInitialLayoutPass();` for reasons like done in avalonia's `DatePicker`:

```csharp
// In `DatePicker.OnFlyoutButtonClicked`:
            // Overlay popup hosts won't get measured until the next layout pass, but we need the
            // template to be applied to `_presenter` now. Detect this case and force a layout pass.
            if (!_presenter.IsMeasureValid)
                (VisualRoot as ILayoutRoot)?.LayoutManager?.ExecuteInitialLayoutPass();
```

Its `internal` now, what's the alternative way for my own custom controls?

**Comment by maxkatz6** [comment, marked as answer]
`_presenter.UpdateLayout()` should do the thing.

**Reply by msneijders** [reply]
I just saw ExecuteInitialLayoutPass is actually doing an ExecuteLayoutPass (UpdateLayout) -- with maybe a measure/arrange before it -- so it should indeed work.
